docs/Deploying/Debugging.md

```diff
@@ -0,0 +1,156 @@
+- fleetctl is getting errors when applying yamls -> `SERVER` 
+- Migrations are taking too long -> `MYSQL`
+- I see 500 errors on the fleetctl or osquery logs, but not on my Fleet logs -> `INFRA`
+
+### SERVER
+
+Whenever diagnosing a server side issue, one of the first steps is to look at Fleet itself. In particular, that means 
+looking at the logs across all instances that are running. How to look at these logs would vary depending on your 
+deployment. If, for instance, it's an AWS deployment, and you're using our terraform files as guidance, you'd use 
+CloudWatch.
+
+Fleet by default will log errors, and those are the first thing to look for. If you have debug logging enabled, you can 
+filter errors by filtering the keyword `err`.
+
+These logs will be the first way to triage a server side error. For example, if there are timeouts happening in APIs, 
+you should continue by looking at `MYSQL` and then `REDIS`. Otherwise, if it looks like a more illustrative error, this 
+would be a good point to reach out with all the information gathered.
+
+If there are no errors in the logs and everything looks normal, check `INFRA`.
+
+### PARTIALSERVER
+
+Sometimes Fleet operates without any errors but accessing a specific part of the web UI are slow. As a starting point it
+would be good to get a screenshot of the Network tab in the Developer Tools of your browser. The main data that needs to 
+be visible are: Name, Status, and Time (in Chrome's terms). 
+[Here's how to accomplish this using Google Chrome](https://developer.chrome.com/docs/devtools/network/).
+
+Besides from this, it might be good to continue with `MYSQL` and `REDIS`.
+
+Depending on the API, there will likely be followup questions about amount of data, but this would be a good point to 
+check in with Engineering.
+
+### MYSQL
+
+Most of the data needed to understand an issue in MySQL should've been gathered already by the basic data specified at 
+the beginning of this document. However, there is a chance that Fleet is running with a database user that is not 
+capable of querying the information needed. So here are the queries that would output a good first step in information
+gathering:
+
+```sql
+show engine innodb status;
+show processlist;
+```
+
+If read replicas are configured, another piece of important data is whether there has been any replication lag 
+registered.
+
+With all this gathered, it's a good time to reach out to the Engineering team.
+
+### REDIS
+
+In most cases, the data gathered at the beginning of this document should be enough to understand what might be 
+happening with Redis. However, if more details are needed, running the 
+[monitor command](https://redis.io/commands/monitor/) should shed more light in the issue.
+
+**WARNING**: if Redis is suffering from performance issues, running monitor will only increase the problem.
+
+### OSQUERY
+
+Just like with the Fleet server, the best way to understand issues on the client side is to look at logs.
```

Comment:
Maybe explain where to look for such logs:
- Orbit: https://github.com/fleetdm/fleet/blob/main/orbit/README.md#logs
- Vanilla osquery: `/var/log/osquery/`